x-pack/platform/plugins/shared/fleet/server/services/package_policy.ts

```diff
@@ -1249,12 +1250,25 @@ class PackagePolicyClientImpl implements PackagePolicyClient {
       this.keepPolicyIdInSync(oldPackagePolicy);
     }
 
+    const agentPolicies: AgentPolicy[] = [];
+
+    for (const policyId of packagePolicyUpdate.policy_ids) {
+      const agentPolicy = await agentPolicyService.get(soClient, policyId, true);
+      if (!agentPolicy) {
+        throw new AgentPolicyNotFoundError('Agent policy not found');
+      }
+
+      agentPolicies.push(agentPolicy);
+    }
+
+    const spaceIds = agentPolicies.flatMap((ap) => ap.space_ids ?? []);
+
```

Comment:
Not totally related to that PR but I am wondering if we need to fetch package policy here (the last flag) it will be great if at some point we have two type for agent policy with and without package policies to be able to validate this with type